src/main/java/htsjdk/samtools/SAMSequenceDictionary.java

```diff
@@ -266,5 +268,100 @@ public String toString() {
+     * values (for tags that are in both) will generate a warning, and the value from dict1 will be used.
+     * For tags that are in tagsToEquate. an unequal value will generate an error (an IllegalArgumentException will
+     * be thrown.)
+     *
+     * @param dict1 first dictionary
+     * @param dict2 first dictionary
+     * @param tagsToMatch list of tags that must be equal if present in both sequence. Typical values will be MD, and LN
+     * @return dictionary consisting of the same sequences as the two inputs with the merged values of tags.
+     */
+    static public SAMSequenceDictionary mergeDictionaries(final SAMSequenceDictionary dict1,
+                                                          final SAMSequenceDictionary dict2,
+                                                          final List<String> tagsToMatch) {
+
+        if(dict1.getSequences().size()!=dict2.getSequences().size()) {
+            throw new IllegalArgumentException(String.format("Do not use this function to merge dictionaries with " +
+                    "different number of sequences in them. Found %d and %d sequences",
+                    dict1.getSequences().size(), dict2.getSequences().size()));
+        }
+        final SAMSequenceDictionary finalDict = new SAMSequenceDictionary();
+        for (final SAMSequenceRecord sequence : dict1.getSequences()) {
+            final SAMSequenceRecord mergedSequence = new SAMSequenceRecord(sequence.getSequenceName(),
+                    UNKNOWN_SEQUENCE_LENGTH);
+            finalDict.addSequence(mergedSequence);
+            final int sequenceIndex = sequence.getSequenceIndex();
+            if (!dict2.getSequence(sequenceIndex).getSequenceName().equals(sequence.getSequenceName())) {
+                throw new IllegalArgumentException(String.format("Non-equal sequence names (%s and %s) at index %d in " +
+                        "the dictionaries.",
+                        sequence.getSequenceName(), dict2.getSequence(sequenceIndex).getSequenceName(), sequenceIndex));
+            }
+
+            final Set<String> allTags = new HashSet<>(dict1.getSequence(sequenceIndex).getAttributes()
+                    .stream().map(Map.Entry::getKey).collect(Collectors.toSet()));
+            allTags.addAll(dict2.getSequence(sequenceIndex).getAttributes()
+                    .stream().map(Map.Entry::getKey).collect(Collectors.toSet()));
+
+            for (final String tag : allTags) {
+                final String value1 = dict1.getSequence(sequenceIndex).getAttribute(tag);
+                final String value2 = dict2.getSequence(sequenceIndex).getAttribute(tag);
+
+                if (value1 != null && value2 != null && !value1.equals(value2)) {
+                    if (tagsToMatch.contains(tag)) {
+                        final String error = String.format("Cannot merge the two dictionaries. Found sequence entry for which " +
```

Comment:
No need for `String.format` here, since log.error() can just be fed a comma separated list of things to concatenate and log.